ops/model.py

```diff
@@ -1326,6 +1328,96 @@ def list_files(self, path: str, *, pattern: str = None,
         """
         return self._pebble.list_files(path, pattern=pattern, itself=itself)
 
+    def push_path(self, source_path: typing.Union[typing.Union[str, Path], typing.List[typing.Union[str, Path]]], dest_dir: typing.Union[str, Path]):
+        source_paths = source_path
+        if not isinstance(source_paths, list):
+            source_paths = [source_paths]
+        source_paths = [str(p) for p in source_paths]
```

Comment:
```suggestion
        source_paths = map(str, [source_path] if not isinstance(source_path, list) else source_path)
```

Reply:
This is more compact, but less legible. I don't object to either one, but I don't think that the golfing is strictly necessary (as fun as it is :-) )

Reply:
functional programmers :P   I'm slightly partial to my dis-functional style.

Reply:
It's probably an unnecessary use of `map` making it look more confusing anyway.

```python
source_paths = source_path if isinstance(list, source_path) else [source_path]
```

Avoids it completely, but it's not really important either way.

I'm more curious about the fact that we're overtly going _away_ from `Path` to `str` instead of the other way around. Using `Path` by default makes any future extensions (like globbing) nicer/easier, and in general lets us trade off verbosity for clarity:

```python
info = os.lstat(path)
if stat.S_ISDIR(info.st_mode):
  ...
  
if stat.S_ISDIR(path.stat().st_mode):
...
```

Though either could be:

```python
if os.path.isdir(fpath):
if os.path.islink(fpath):
if os.path.isfile(fpath):

if fpath.isdir():
...
```

It's probably worth defining a type near the top of the file to avoid the amount of repetition in `typing.Union[str, Path]` either way. The indentation looks a little funky here to keep pep8 happy.

Reply:
I've switched things to pathlib.